src/gt4py/next/common.py

```diff
@@ -613,3 +654,26 @@ def promote_dims(*dims_list: Sequence[Dimension]) -> list[Dimension]:
         )
 
     return topologically_sorted_list
+
+
+class FieldBuiltinFuncRegistry:
+    _builtin_func_map: collections.ChainMap[
+        fbuiltins.BuiltInFunction, Callable
+    ] = collections.ChainMap()
+
+    def __init_subclass__(cls, **kwargs):
+        # might break in multiple inheritance (if multiple ancestors have `_builtin_func_map`)
+        cls._builtin_func_map = cls._builtin_func_map.new_child()
```

Comment:
What's the `[0]` in `c.__dict__["_builtin_func_map"][0]`?

Reply:
The `[0]` means that only the first `dict` of the chainmap is taken, which is the `dict` added in that specific superclass. Taken the full chainmap of every superclass would create a wrong chainmap, because we will be concatenating the MROs of the superclasses instead of building up a chainmap with the dicts from the classes in the MRO of the new class.

Reply:
Oh, makes sense.

Reply:
Should I take it in this PR?

Reply:
Let's do it in a different one.